README.md

```diff
@@ -193,3 +224,107 @@ def my_awesome_check(
 ```
+
+## Benchmark plugins
+
+Any project can add benchmarks to `rapids benchmark` by exposing a function via an entrypoint, similar to how
+check plugins work for `rapids doctor`.
+
+Here's a simple example benchmark function:
+
+```python
+# src/benchmark_example/benchmark.py
+import time
+import numpy as np
+
+
+def my_array_benchmark(verbose=False):
+    """Benchmark array operations comparing NumPy vs CuPy."""
+    try:
+        import numpy as np
+    except ImportError as e:
+        raise ImportError("numpy is required for this benchmark") from e
+
+    try:
+        import cupy as cp
+    except ImportError as e:
+        raise ImportError("cupy is required for GPU benchmarking") from e
+
+    # Prepare data
+    size = 10_000_000
+    data = np.random.randn(size).astype(np.float32)
+
+    if verbose:
+        print(f"Running array operations on {size:,} elements")
+
+    # CPU timing
+    start = time.time()
+    cpu_array = np.array(data)
+    result_cpu = np.sum(cpu_array**2) + np.mean(cpu_array)
+    cpu_time = time.time() - start
+
+    # GPU timing (excluding data transfer)
+    gpu_array = cp.array(data)  # Transfer data (not timed)
+    _ = len(gpu_array)  # GPU warmup
+    start = time.time()
+    result_gpu = cp.sum(gpu_array**2) + cp.mean(gpu_array)
+    gpu_time = time.time() - start
+
+    if verbose:
+        print(f"CPU: {cpu_time:.3f}s, GPU: {gpu_time:.3f}s")
+
+    return cpu_time, gpu_time
+```
+
+Register this function by adding an entrypoint to the `rapids_benchmark` group in `pyproject.toml`:
+
+```toml
+[project.entry-points.rapids_benchmark]
+array_operations = "benchmark_example.benchmark:my_array_benchmark"
+```
+
+### Benchmark Options
+
+The benchmark command supports several options to customize execution:
+
+```console
+$ rapids benchmark --help
+
+ Usage: rapids benchmark [OPTIONS] [FILTERS]...
+
+ Run performance benchmarks comparing CPU vs GPU implementations.
+
+╭─ Options ───────────────────────────────────────────────────────────────────────────╮
+│ --verbose     Enable verbose mode for detailed timing output.                       │
+│ --dry-run     Perform a dry run without running benchmarks.                         │
+│ --runs        Number of iterations to run each benchmark (default: 5).              │
+│ --help        Show this message and exit.                                           │
+╰─────────────────────────────────────────────────────────────────────────────────────╯
+```
+
+**Examples:**
+
+```console
+# Run with custom number of iterations
+$ rapids benchmark --runs 10
+
+# Run only specific benchmarks
+$ rapids benchmark dataframe
+
+# See detailed timing information
+$ rapids benchmark --verbose
+
+# Check what benchmarks are available without running them
+$ rapids benchmark --dry-run --verbose
+```
+
+### Benchmark function guidelines
+
+Benchmark functions should follow this pattern:
+
+1. **Function signature**: Accept `**kwargs` for forward compatibility
```

Comment:
Can benchmarks be parametrized? We very frequently need to scale benchmarks up or down to fit well for a given device, or do scaling studies, or study how data type (int, float, etc.) affects performance.